### ISSUE TITLE
Conflict with SilverStripeNavigator when archiving a page in a non-standard locale

### ISSUE THREAD

**Issue by robbieaverill** [tractorcow/silverstripe-fluent#219]
Hey Damian,

I've just added a new locale to our configuration (en_US) and ran a test by changing to that locale (so that the `l=en_US` param is added to the query string), then archiving the page.

It looks like `SilverStripeNavigator` (CMS module) is adding `?archiveDate=...` to the querystring manually, rather than doing it intelligently to see what's already in the querystring, e.g. I end up hitting this: `/content/preview?page_id=28&l=en_US?archiveDate=2016-06-24+04%3A48%3A09&CMSPreview=1`

The stack trace is:

```
[User Error] Uncaught BadMethodCallException: Invalid locale en_US?archiveDate=2016-06-24 04:48:09
GET /content/preview?page_id=28&l=en_US?archiveDate=2016-06-24+04%3A48%3A09&CMSPreview=1

Line 620 in /var/www/instances/silverstripe-dev/fluent/code/Fluent.php
Source

611      */
612     public static function with_locale($locale, $callback)
613     {
614 
615         // Check and set locale
616         if (self::$_override_locale) {
617             throw new BadMethodCallException("Fluent::with_locale cannot be nested");
618         }
619         if (!in_array($locale, self::locales())) {
620             throw new BadMethodCallException("Invalid locale $locale");
621         }
622         self::$_override_locale = $locale;
623         DataObject::flush_and_destroy_cache();
624 
625         // Callback
626         $result = call_user_func($callback);

Trace

    Fluent::with_locale(en_US?archiveDate=2016-06-24 04:48:09,Closure)
    MyContentController.php:36
    MyContentController->preview(SS_HTTPRequest)
    RequestHandler.php:288
    RequestHandler->handleAction(SS_HTTPRequest,preview)
    Controller.php:202
    Controller->handleAction(SS_HTTPRequest,preview)
    RequestHandler.php:200
    ...
```

I'm not necessarily opposed to logging this under the CMS module and taking a look at it, if you'd rather.

Some background:

* We're using a custom preview controller, which acts as a PHP proxy to an external server to render the preview HTML
* We do use query string parameters in this controller: page ID, as well as the stage which I assume is normal for SilverStripe

Happy to discuss!

**Comment by robbieaverill** [comment]
Fixed in silverstripe/silverstripe-cms#1558

**Comment by tractorcow** [comment]
Well, it WILL be fixed once it's merged. :)

**Comment by robbieaverill** [comment]
Fixed in framework/CMS 3.4.1 :) thanks guys!
